web/crux/src/app/deploy/deploy.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import PrismaService from 'src/services/prisma.service'
+
+@Injectable()
+export default class DeployRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async findDeploymentEventsById(deploymentId: string): Promise<any> {
+    const deploymentEvents = await this.prisma.deploymentEvent.findMany({
+      where: {
+        deploymentId,
+      },
+    })
+
+    if (deploymentEvents.length === 0) {
```

Comment:
I think this is unnecessary. A simple empty array is sufficent.

Reply:
It's more explicit this way, but I can remove.

Reply:
I believe we didn't throw any error on other list-returning endpoints for an empty array. Remove it please.